### ISSUE TITLE
Bulk mode does not work when using @requestbody 

### ISSUE THREAD

**Issue by ebann** [wnameless/spring-bulk-api#4]
Something like this fails:
@Bulkable
public class MyController{

class MyInnerInfos {
String c;
}

class MyInfos {
MyInnerInfos a;
String b;
}

  @RequestMapping(value = "list", method = RequestMethod.POST)
  public MyResponse list(@RequestBody MyInfos infos){
 ...
}
}

It seems it's caused by the implementation of DefaultBulkApiService#requestEntity implementation.
It uses a LinkedMultiValueMap which leads to a serialization of the json to something like:

{a=[{c=...}], b=[...]}

Which is not a valid input for the above controller.

Replacing LinkedMultiValueMap by a regular Map when one of the given parameter is itself a complex structure (Map)

To summarize:
This works:

  private RequestEntity<?> requestEntity(BodyBuilder bodyBuilder, BulkOperation op) {
    for (Entry<String, String> header : op.getHeaders().entrySet()) {
      bodyBuilder.header(header.getKey(), header.getValue());
    }

    Object params;
    if (op.getParams().values().stream().anyMatch(Map.class::isInstance)) {
      params = op.getParams();
    }
    else {
      params = new LinkedMultiValueMap<String, Object>();
      ((MultiValueMap) params).setAll(op.getParams());
    }
    return bodyBuilder.body(params);
  }

**Comment by wnameless** [comment]
Thanks for the patch. However, it needs more works to solve the problem perfectly. I just released a new version v0.6.1 to fix this problem. Please take a try.

**Comment by ebann** [comment]
Thanks, 6.1 works for my @requestbody case and your implementation is way cleaner :)

Thanks
